extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -221,40 +228,156 @@ class AmpAccordion extends AMP.BaseElement {
     const content = sectionComponents[1];
     const contentId = content.getAttribute('id');
     const isSectionClosedAfterClick = section.hasAttribute('expanded');
-    this.mutateElement(() => {
-      let toExpand;
-      if (opt_forceExpand === true) {
-        toExpand = true;
-      } else if (opt_forceExpand === false) {
-        toExpand = false;
-      } else {
-        toExpand = !section.hasAttribute('expanded');
-      }
+    let toExpand;
+    if (opt_forceExpand === true) {
+      toExpand = true;
+    } else if (opt_forceExpand === false) {
+      toExpand = false;
+    } else {
+      toExpand = !section.hasAttribute('expanded');
+    }
 
+    // Animate Toggle
+    if (this.element.hasAttribute('animate')) {
       if (toExpand) {
-        section.setAttribute('expanded', '');
         header.setAttribute('aria-expanded', 'true');
-        // if expand-single-section is set, only allow one <section> to be
-        // expanded at a time
+        this.animateExpand_(section).then(() => {
+        });
         if (this.element.hasAttribute('expand-single-section')) {
           this.sections_.forEach(sectionIter => {
             if (sectionIter != section) {
-              sectionIter.removeAttribute('expanded');
+              this.animateCollapse_(sectionIter);
               sectionIter.children[0].setAttribute('aria-expanded', 'false');
             }
           });
         }
       } else {
-        section.removeAttribute('expanded');
         header.setAttribute('aria-expanded', 'false');
+        this.animateCollapse_(section);
       }
-    }, section);
+    } else { // Toggle without animation
+      this.mutateElement(() => {
+        if (toExpand) {
+          section.setAttribute('expanded', '');
+          header.setAttribute('aria-expanded', 'true');
+          // if expand-single-section is set, only allow one <section> to be
+          // expanded at a time
+          if (this.element.hasAttribute('expand-single-section')) {
+            this.sections_.forEach(sectionIter => {
+              if (sectionIter != section) {
+                sectionIter.removeAttribute('expanded');
+                sectionIter.children[0].setAttribute('aria-expanded', 'false');
+              }
+            });
+          }
+        } else {
+          section.removeAttribute('expanded');
+          header.setAttribute('aria-expanded', 'false');
+        }
+      }, section);
+    }
     this.currentState_[contentId] = !isSectionClosedAfterClick;
     this.setSessionState_();
   }
 
+  /**
```

Comment:
Two reasons for using the animations library. 
1. Allows us to expose expand and collapse events after the animation actually happens. I originally thought we needed this for collapsing children, but it turns out that approach does not work. Nevertheless, it's still possible that we would want to expose these events eventually at some point in time. 
2. Allows us to vary duration based on height.